### PR TITLE
cuda: add option to remove warning about missing libcuda.

### LINKED ISSUE #4405: Suppress warning about missing `libcuda.so.1`
I've built OpenMPI v3.0.0 with CUDA support. We have some hardware which has GPUs and some that does not. Fortunately, if `libcuda.so.1` cannot be found OpenMPI will still work. However, it prints this message:

```
The library attempted to open the following supporting CUDA libraries,
but each of them failed.  CUDA-aware support is disabled.
libcuda.so.1: cannot open shared object file: No such file or directory
libcuda.dylib: cannot open shared object file: No such file or directory
/usr/lib64/libcuda.so.1: cannot open shared object file: No such file or directory
/usr/lib64/libcuda.dylib: cannot open shared object file: No such file or directory
If you are not interested in CUDA-aware support, then run with
--mca mpi_cuda_support 0 to suppress this message.  If you are interested
in CUDA-aware support, then try setting LD_LIBRARY_PATH to the location
of libcuda.so.1 to get passed this issue.
```

I don't want to put `mpi_cuda_support 0` in my global conf file because I do want CUDA support, just only when CUDA is available. But I don't want this awful message for non-CUDA users.

Is there an environment variable that corresponds to `mpi_cuda_support`? Alternatively is there an option that simply suppresses the warning *but not the behavior*? If not it seems like we should add a knob somewhere for this.

### PR DESCRIPTION
**Pull request by sjeaugey**
Closes #4405

### CODE REVIEW

File: opal/mca/common/cuda/common_cuda.c [hunk 1]
```diff
@@ -429,8 +429,10 @@ int mca_common_cuda_stage_one_init(void)
 
     if (true != stage_one_init_passed) {
         errmsg = opal_argv_join(errmsgs, '\n');
-        opal_show_help("help-mpi-common-cuda.txt", "dlopen failed", true,
-                       errmsg);
+        if (opal_warn_on_missing_libcuda) {
+          opal_show_help("help-mpi-common-cuda.txt", "dlopen failed", true,
```
Comment (jsquyres) [inline opal/mca/common/cuda/common_cuda.c]: Super minor nit: should be a 4-space tab, not a 2-space tab.

**Comment by sjeaugey** [comment]
@jsquyres spaces fixed and commits squashed.

**Comment by jsquyres** [comment]
bot:ompi:retest

**Comment by jsquyres** [comment]
bot:ompi:retest

**Comment by jsquyres** [comment]
bot:ompi:retest

**Comment by jsquyres** [comment]
@sjeaugey You might want to see if the RM's are amenable to you cherry picking this PR to the release branches.

**Pull request merged**